### PR TITLE
[lang] Fix scalarization for PrintStmt

### PR DESCRIPTION
**Pull request by jim19930609**
Issue: https://github.com/taichi-dev/taichi/issues/6927

### CODE REVIEW

**Comment by neozhaoliang** [comment]
There is a redundant `, ` before each ']'?

```python
import taichi as ti

ti.init()

@ti.kernel
def test():
    v = ti.Matrix([[1., 2.], [3., 4.]])
    print(v)

test()
```

The output:
```
[[1.000000, 2.000000, ], [3.000000, 4.000000, ], ]
```

File: taichi/transforms/scalarize.cpp [hunk 1]
```diff
@@ -281,15 +281,55 @@ class Scalarize : public BasicStmtVisitor {
...
+          int n = is_matrix ? tensor_shape[1] : 1;
+
+          new_contents.push_back("[");
+          if (is_matrix) {
+            for (size_t i = 0; i < m; i++) {
+              new_contents.push_back("[");
+              for (size_t j = 0; j < n; j++) {
+                size_t index = i * n + j;
+                new_contents.push_back(matrix_init_stmt->values[index]);
+                new_contents.push_back(", ");
```
Comment (neozhaoliang) [inline taichi/transforms/scalarize.cpp:298]: I believe here we should check if we are processing the last element?

Comment (neozhaoliang) [inline taichi/transforms/scalarize.cpp:298]:
```suggestion
    if (j != n - 1)
                new_contents.push_back(", ");
```
I believe here we should check if we are processing the last element?

File: taichi/transforms/scalarize.cpp [hunk 2]
```diff
@@ -281,15 +281,56 @@ class Scalarize : public BasicStmtVisitor {
...
         if (print_stmt->is<MatrixInitStmt>()) {
           auto matrix_init_stmt = print_stmt->cast<MatrixInitStmt>();
-          for (size_t j = 0; j < matrix_init_stmt->values.size(); j++) {
-            new_contents.push_back(matrix_init_stmt->values[j]);
+          auto tensor_shape =
+              print_stmt->ret_type->as<TensorType>()->get_shape();
+
+          bool is_matrix = tensor_shape.size() == 2;
+          int m = tensor_shape[0];
+          int n = is_matrix ? tensor_shape[1] : 1;
```
Comment (strongoier) [inline taichi/transforms/scalarize.cpp:289]: nit: `n` can be defined inside the `is_matrix` branch below.

File: taichi/transforms/scalarize.cpp [hunk 3]
```diff
@@ -281,15 +281,56 @@ class Scalarize : public BasicStmtVisitor {
...
+          if (is_matrix) {
+            for (size_t i = 0; i < m; i++) {
+              new_contents.push_back("[");
+              for (size_t j = 0; j < n; j++) {
+                size_t index = i * n + j;
+                new_contents.push_back(matrix_init_stmt->values[index]);
+                if (j != n - 1)
+                  new_contents.push_back(", ");
+              }
+              new_contents.push_back("], ");
```
Comment (strongoier) [inline taichi/transforms/scalarize.cpp:301]: nit: `,` should not be inserted when `i == m - 1`.

File: taichi/transforms/scalarize.cpp [hunk 4]
```diff
@@ -281,15 +281,56 @@ class Scalarize : public BasicStmtVisitor {
...
+            }
+          } else {
+            for (size_t i = 0; i < m; i++) {
+              new_contents.push_back(matrix_init_stmt->values[i]);
+              new_contents.push_back(", ");
```
Comment (strongoier) [inline taichi/transforms/scalarize.cpp:306]: ditto

**Review by neozhaoliang — APPROVED** [review]
LGTM!

**Review by strongoier — APPROVED** [review]

**Pull request merged**